volatility/framework/plugins/windows/netlist.py

```diff
@@ -0,0 +1,544 @@
+                       tcpip_symbol_table: str,
+                       tcpip_module_offset: int,
+                       pointer_length: int) -> (int, int):
+        """Finds the given image's port pools. Older Windows versions (presumably < Win10 build 14251) use driver
+        symbols called `UdpPortPool` and `TcpPortPool` which point towards the pools.
+        Newer Windows versions use `UdpCompartmentSet` and `TcpCompartmentSet`, which we first have to translate into
+        the port pool address. See also: http://redplait.blogspot.com/2016/06/tcpip-port-pools-in-fresh-windows-10.html
+
+        Args:
+            context: The context to retrieve required elements (layers, symbol tables) from
+            layer_name: The name of the layer on which to operate
+            net_symbol_table: The name of the table containing the tcpip types
+            tcpip_module_offset: This memory dump's tcpip.sys image offset
+            tcpip_symbol_table: The name of the table containing the tcpip driver symbols
+            pointer_length: Length of this architecture's pointers
+
+        Returns:
+            The tuple containing the address of the UDP and TCP port pool respectively.
+        """
+
+        if "UdpPortPool" in context.symbol_space[tcpip_symbol_table].symbols:
+            # older Windows versions
+            upp_symbol = context.symbol_space.get_symbol(tcpip_symbol_table + constants.BANG + "UdpPortPool").address
+            upp_addr = cls.read_pointer(context, layer_name, tcpip_module_offset + upp_symbol, pointer_length)
+
+            tpp_symbol = context.symbol_space.get_symbol(tcpip_symbol_table + constants.BANG + "TcpPortPool").address
+            tpp_addr = cls.read_pointer(context, layer_name, tcpip_module_offset + tpp_symbol, pointer_length)
+
+        elif "UdpCompartmentSet" in context.symbol_space[tcpip_symbol_table].symbols:
+            # newer Windows versions since 10.14xxx
+            ucs = context.symbol_space.get_symbol(tcpip_symbol_table + constants.BANG + "UdpCompartmentSet").address
+            tcs = context.symbol_space.get_symbol(tcpip_symbol_table + constants.BANG + "TcpCompartmentSet").address
+
+            ucs_offset = cls.read_pointer(context, layer_name, tcpip_module_offset + ucs, pointer_length)
+            tcs_offset = cls.read_pointer(context, layer_name, tcpip_module_offset + tcs, pointer_length)
+
+            ucs_obj = context.object(net_symbol_table + constants.BANG + "_INET_COMPARTMENT_SET", layer_name = layer_name, offset = ucs_offset)
+            upp_addr = ucs_obj.InetCompartment.dereference().ProtocolCompartment.dereference().PortPool
+
+            tcs_obj = context.object(net_symbol_table + constants.BANG + "_INET_COMPARTMENT_SET", layer_name = layer_name, offset = tcs_offset)
+            tpp_addr = tcs_obj.InetCompartment.dereference().ProtocolCompartment.dereference().PortPool
+
+        else:
+            # this branch should not be reached.
```

Comment:
Thanks! Was not sure which exception to use as `SymbolError` only takes one symbol in its main argument but I think it works out here.

Reply:
Yeah, I think that's the right choice, and just putting one of the symbols would be the equivalent of raising twice with one of them first, so that's fine too.  5:)